### PR TITLE
EdgeClass

### PR DESCRIPTION
**Pull request by LeoWikstrom**
Added class for edges between room. This class keeps track of which edges are connected, and which entities and NavTriangles belong to each edge. To test, insert the following code:

In the main loop, add:

	static int toRemove = 0;
        static bool pressedEnter = false;

        if (Input::GetInstance().GetKeyState(SCAN_CODES::ENTER))
        {
            pressedEnter = true;
        }
        if (!Input::GetInstance().GetKeyState(SCAN_CODES::ENTER) && pressedEnter)
        {
            pressedEnter = false;
            AssetLoader::Get()->GetEdges().at(toRemove)->RemoveEntitiesFromWorld();
            toRemove = (toRemove + 1) % AssetLoader::Get()->GetEdges().size();
        }

Where the scene is loaded in main (where the function LoadMap is called), replace the LoadMap call with the following:
	al->LoadMap(scene, "../Vendor/Resources/BaseRoom.map", 0);
    	al->LoadMap(scene, "../Vendor/Resources/ExampleRoom.map", 1, { 173.0, 0.0, 0.0 });
    	al->LoadMap(scene, "../Vendor/Resources/ExampleRoom.map", 2, { 86.5, 0.0, 150.0 });
    	al->LoadMap(scene, "../Vendor/Resources/ExampleRoom.map", 3, { -173.0, 0.0, 0.0 });
    	al->LoadMap(scene, "../Vendor/Resources/ExampleRoom.map", 4, { -259.5, 0.0, 150.0 });
    	al->LoadMap(scene, "../Vendor/Resources/ExampleRoom.map", 5, { -173.0, 0.0, 300.0 });
    	al->LoadMap(scene, "../Vendor/Resources/ExampleRoom.map", 6, { 0.0, 0.0, 300.0 });

In GameInitScene() add:

    std::vector<Edge*> edges = AssetLoader::Get()->GetEdges();
    edges.at(0)->RemoveEntitiesFromWorld();
    edges.at(1)->RemoveEntitiesFromWorld();
    edges.at(4)->RemoveEntitiesFromWorld();
    edges.at(10)->RemoveEntitiesFromWorld();
    edges.at(11)->RemoveEntitiesFromWorld();
    edges.at(14)->RemoveEntitiesFromWorld();
    edges.at(15)->RemoveEntitiesFromWorld();
    edges.at(17)->RemoveEntitiesFromWorld();
    edges.at(19)->RemoveEntitiesFromWorld();
    edges.at(23)->RemoveEntitiesFromWorld();
    edges.at(24)->RemoveEntitiesFromWorld();
    edges.at(26)->RemoveEntitiesFromWorld();
    edges.at(31)->RemoveEntitiesFromWorld();
    edges.at(33)->RemoveEntitiesFromWorld();
    edges.at(38)->RemoveEntitiesFromWorld();
    edges.at(40)->RemoveEntitiesFromWorld();

### CODE REVIEW

File: StortSpelProjekt/Engine/src/Renderer/Renderer.cpp [hunk 1]
```diff
@@ -2057,7 +2057,7 @@ void Renderer::initRenderTasks()
 	gpsdShadow.SampleMask = UINT_MAX;
 	// Rasterizer behaviour
 	gpsdShadow.RasterizerState.FillMode = D3D12_FILL_MODE_SOLID;
-	gpsdShadow.RasterizerState.CullMode = D3D12_CULL_MODE_BACK;
+	gpsdShadow.RasterizerState.CullMode = D3D12_CULL_MODE_NONE;
```
Comment (Schtekt) [inline StortSpelProjekt/Engine/src/Renderer/Renderer.cpp]: Backface culling should not be off right?

File: StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp [hunk 2]
```diff
@@ -141,6 +141,8 @@ void component::RangeComponent::Attack()
 		pos.y = ParentPos.y + (forward.y / length);
 		pos.z = ParentPos.z + (forward.z / length) * (dim.z * scale.z / 2.0) * 1.1f;
 
+		Log::Print("Playerpos: %f, %f, %f\n", ParentPos.x, ParentPos.y, ParentPos.z);
+
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/RangeComponent.cpp]: This is unneccessary, right?

**Review by Schtekt — CHANGES_REQUESTED** [review]
Looks good!
A few minor things I would like to change or take a further gander at.

**Review by Schtekt — DISMISSED** [review]

**Review by kattalainen — DISMISSED** [review]
This rar file is probably not needed: StortSpelProjekt/Vendor/Resources/Models/Torch.rar.
Other than that it's looking good!

**Review by Schtekt — APPROVED** [review]

**Review by kattalainen — APPROVED** [review]

**Pull request merged**